### PR TITLE
[SC-32565] Setup publishing on NPM

### PR DESCRIPTION
**Pull request by nrutman**
This PR:

* Configures the `build` script for packaging
* Does minor updates to bring dependencies current
* Creates a `publish` workflow file

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,6 +17,9 @@
     "type": "git",
     "url": "git+https://github.com/fieldguide/pipeline.git"
   },
+  "files": [
+    "build/"
```
Comment (namoscato) [inline package.json:21]: Minor, but this contains the compiled tests at the moment. I'd recommend the following:

1. Create a separate TSConfig for pre-publishing, i.e. `tsconfig.pack.json`:

    ```json
    {
      "extends": "./tsconfig.json",
      "exclude": ["**/__mocks__", "**/__tests__"]
    }
    ```

1. Update the build script to target that configuration:

    ```bash
    tsc --project tsconfig.pack.json
    ```

File: package.json [hunk 2]
```diff
@@ -1,7 +1,9 @@
 {
   "name": "@fieldguide/pipeline",
   "description": "A toolkit to easily build synchronous process pipelines in TypeScript/JavaScript",
-  "version": "0.1.0",
+  "version": "0.0.1",
```
Comment (namoscato) [inline package.json:4]: Minor, but if you run `npm i` again, this version change will update `package-lock.json`.

**Review by namoscato — CHANGES_REQUESTED** [review]
Just a few minor things.

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
+name: "Publish to NPM"
```
[the remark this answers is not in the document]
Comment (nrutman) [reply .github/workflows/publish.yml]: 1b12284

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+          node-version-file: ".nvmrc"
+          cache: "npm"
+          registry-url: "https://registry.npmjs.org"
+          scope: "@fieldguide"
+
+      - name: Install dependencies
+        run: npm ci --prefer-offline --no-audit
+
+      - name: Build code
+        run: npm run build
```
[the remark this answers is not in the document]
Comment (nrutman) [reply .github/workflows/publish.yml]: Updated as a part of 1b12284. `prepack` now runs `build`.

File: package.json (hunk 1 above)
Comment (nrutman) [reply package.json:22]: This is great, and included in 1b12284. I used `tsconfig.build.json` to match the main command name. To me, `pack` feels like a very npm-kind-of-thing. `build` feels more generalized. So the mental model I took is that the npm `prepack` stage should run the `build`.

Super small, and at the end of the day it has little consequence. I don't know why I have opinions about these kinds of things, but I do. 😆

File: package.json (hunk 2 above)
Comment (nrutman) [reply package.json:4]: Great catch. I've updated it in 1b12284.

**Comment by nrutman** [comment]
@namoscato I went through and noted that all of the requested changes should be updated in 1b12284. If you get a chance, can you review this again? Thanks!

File: package.json (hunk 1 above)
Comment (namoscato) [reply package.json:22]: Yeah, "build" is a better word 😅

**Review by namoscato — APPROVED** [review]
:shipit:

**Pull request merged**